clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -272,6 +272,7 @@ private boolean trySend(long now) {
 
     private void clientPoll(long timeout, long now) {
         client.poll(timeout, now);
+        now = time.milliseconds();
```

Comment:
I like this idea also, though I feel a little uneasy that the burden is still on the caller to ensure that the cached value is current on method invocation. There would be nothing in the signature to suggest that it's needed except javadoc. Maybe it would be clearer if instead of Time, the class is called Now?